### ISSUE TITLE
Grafana.ini Configmap role_attribute_path

### ISSUE THREAD

**Issue by gruppferi** [grafana/helm-charts#270]
Deploying grafana with auth.generic_oauth working as far as I don't use the role_attribute_path.
I tried in quotation and without quotation no lack.
`
  auth.generic_oauth:
    enabled: true
    client_id: "$__file{/etc/secrets/auth_generic_oauth/client_id}"
    client_secret: "$__file{/etc/secrets/auth_generic_oauth/client_secret}"
    auth_url: "{{ keycloak.publicUrl }}/auth/realms/xxx/protocol/openid-connect/auth"
    token_url: "{{ keycloak.publicUrl }}/auth/realms/xxx/protocol/openid-connect/token"
    api_url: "{{ keycloak.publicUrl }}/auth/realms/xxx/protocol/openid-connect/userinfo"
    allow_sign_up: true
    scopes: "openid profile email"
    role_attribute_path: "contains(roles[*], 'admin') && 'Admin' || contains(roles[*], 'editor') && 'Editor' || 'Viewer'"
`

From the last version which I used (stable repo helm) grafana to this 7.4.2 the only change in configmap I see is `{{- if kindIs "invalid" $elemVal }}` added.

**Comment by diranged** [comment]
So I just ran into this bug and couldn't figure out why `helm template ...` worked fine on my local dev, but failed in ArgoCD. After digging, I discovered that ArgoCD (1.8.5) uses helm 3.4.1:

```
$ ../darwin-amd64/helm version
version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"clean", GoVersion:"go1.14.11"}
$ ../darwin-amd64/helm template . --name-template observability --namespace observability --values values.yaml 
coalesce.go:199: warning: destination for tlsConfig is a table. Ignoring non-table value <nil>
coalesce.go:199: warning: destination for tlsConfig is a table. Ignoring non-table value <nil>
coalesce.go:199: warning: destination for tlsConfig is a table. Ignoring non-table value <nil>
Error: YAML parse error on .../charts/kube-prometheus-stack/charts/grafana/templates/configmap.yaml: error converting YAML to JSON: yaml: line 36: did not find expected alphabetic or numeric character
```

If you run this same command with helm 3.5.2, it works fine.. so there is some incompatibility with the functions being used here. Given that this breaks deployments in popular tools like ArgoCD, can we consider reverting this change for now?

**Comment by diranged** [comment]
I was able to work around this for now by removing the ` ... || ...` and exclusively writing a role_attribution_Path jmes query that only has one match.. but thats not a long term fix.

**Comment by gruppferi** [comment]
@diranged  Yes I Found out the same later that night, I was using helm 3.0.2 after upgrading to 3.5.2 it worked fine. I also exactly came to same idea removing spaces before helm issue which worked but solution is upgrading to latest Helm

**Comment by valencetech** [comment]
no longer an issue with Helm v3.5.3

**Comment by diranged** [comment]
Unfortunately I don't believe that upgrading helm is the right answer for users here. Many users are using tools like Flux/Argo that have baked in Helm. Asking them to upgrade Helm is a significant chore in that case.  Can we revert or update this change so that it just works?

**Comment by chopmann** [comment]
We are hitting this bug with helm-operator, which is being deprecated. Is there a workaround?

**Comment by Unibond7** [comment]
We're hitting this issue when trying to deploy [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/880c706a26a9a6224236275d73a29681048f008c/charts/kube-prometheus-stack), which this chart is a dependency of, via the Terraform helm provider (which currently is set to Helm version 3.3.4), so upgrading to a newer version of helm is not a viable option.

@diranged could you share your jmes query as the interim solution. See if we can implement this in our terraform code.

Currently we are disabling installing grafana with the kube-prometheus-stack chart and deploying grafana independently,  pinning the chart version to 6.4.4, as that's the latest version were we don't see this error. This though means we don't get the default dashboards that charts comes with, which are a nice to have.

**Comment by diranged** [comment]
> We're hitting this issue when trying to deploy [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/880c706a26a9a6224236275d73a29681048f008c/charts/kube-prometheus-stack), which this chart is a dependency of, via the Terraform helm provider (which currently is set to Helm version 3.3.4), so upgrading to a newer version of helm is not a viable option.
> 
> @diranged could you share your jmes query as the interim solution. See if we can implement this in our terraform code.
> 
> Currently we are disabling installing grafana with the kube-prometheus-stack chart and deploying grafana independently, pinning the chart version to 6.4.4, as that's the latest version were we don't see this error. This though means we don't get the default dashboards that charts comes with, which are a nice to have.

```yaml
#role_attribute_path: contains(groups[*], 'grafana-admins') && 'Admin' || contains(groups[*], 'grafana-editors') && 'Editor' || 'Viewer'
role_attribute_path: contains(groups[*], 'grafana-admins') && 'Admin'
```

**Comment by cmanzur** [comment]
The solution is wrapping using "()". Like this:
`"(contains(roles[*], 'admin') && 'Admin') || (contains(roles[*], 'editor') && 'Editor')`

**Comment by R-Studio** [comment]
Same issue here, but even when I remove the "| |" it does not work and through following error:
`Error: UPGRADE FAILED: error validating "": error validating data: ValidationError(Grafana.spec.config.auth.github): unknown field "role_attribute_path" in org.integreatly.v1alpha1.Grafana.spec.config.auth.github`

If I use `helm template` instead of `helm upgrade --install`, it works.
